<img src="images/usm.jpg" width="480" height="240" align="left"/>

# MAT281 - Laboratorio N°03

## Objetivos de la clase

* Reforzar los conceptos básicos de pandas.

## Contenidos

* [Problema 01](#p1)


## Problema 01


<img src="https://imagenes.universia.net/gc/net/images/practicas-empleo/p/pr/pro/profesiones-con-el-avance-de-la-tecnologia.jpg" width="480" height="360" align="center"/>


EL conjunto de datos se denomina `ocupation.csv`, el cual contiene información tal como: edad ,sexo, profesión, etc.

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [1]:
import pandas as pd
import os

In [2]:
# cargar datos
df = pd.read_csv(os.path.join("data","ocupation.csv"), sep="|").set_index('user_id')
df.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

In [41]:
# se eliminan los valore nulos del dataframe
df = df[lambda df: df.notnull().all(axis=1)]

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:

1. ¿Cuál es el número de observaciones en el conjunto de datos?

In [42]:
df.shape

(943, 4)

Por lo tanto el número de observaciones es 943 

2. ¿Cuál es el número de columnas en el conjunto de datos?

In [43]:
number = len(df.columns)
print("el número de columnas en el conjunto de datos es:",number)

el número de columnas en el conjunto de datos es: 4


3. Imprime el nombre de todas las columnas

In [44]:
for i in range(len(df.columns)):
    print(df.columns[i])

age
gender
occupation
zip_code


4. Imprima el índice del dataframe

In [45]:
df.index

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

5. ¿Cuál es el tipo de datos de cada columna?

In [53]:
print("Tipo de dato por columna:")
df.dtypes 

Tipo de dato por columna:


age            int64
gender        object
occupation    object
zip_code      object
dtype: object

6. Resumir el conjunto de datos

In [47]:
df.describe()

age
count  943.000000
mean    34.051962
std     12.192740
min      7.000000
25%     25.000000
50%     31.000000
75%     43.000000
max     73.000000

7. Resume conjunto de datos  con todas las columnas

In [48]:
df.describe(include='all')

age gender occupation zip_code
count   943.000000    943        943      943
unique         NaN      2         21      795
top            NaN      M    student    55414
freq           NaN    670        196        9
mean     34.051962    NaN        NaN      NaN
std      12.192740    NaN        NaN      NaN
min       7.000000    NaN        NaN      NaN
25%      25.000000    NaN        NaN      NaN
50%      31.000000    NaN        NaN      NaN
75%      43.000000    NaN        NaN      NaN
max      73.000000    NaN        NaN      NaN

8. Imprimir solo la columna de **occupation**.

In [49]:
df['occupation']

user_id
1         technician
2              other
3             writer
4         technician
5              other
           ...      
939          student
940    administrator
941          student
942        librarian
943          student
Name: occupation, Length: 943, dtype: object

9. ¿Cuántas ocupaciones diferentes hay en este conjunto de datos?

In [111]:
ocupaciones_diferentes = df['occupation'].unique()
ocupaciones_totales = list(ocupaciones_diferentes)
n = len(ocupaciones_totales)
print("el número de ocupaciones diferentes es:" ,n)

el número de ocupaciones diferentes es: 21


10. ¿Cuál es la ocupación más frecuente?

In [112]:
diccionario={}

for ocupacion in ocupaciones_diferentes:
    diccionario[ocupacion]=0

for ocupacion1 in diccionario.keys():
    for ocupacion2 in df['occupation']:
        if ocupacion1==ocupacion2:
            diccionario[ocupacion1]+=1 

dataframe = pd.DataFrame({
            "ocupacion":diccionario.keys(),
             "ocurrencia":diccionario.values()
})

dataframe

maxima_frecuencia=dataframe['ocurrencia'].max()

for i in diccionario.keys():
    if diccionario[i]==maxima_frecuencia:
        print("La ocupación con la mayor ocurrencia es:",i)

La ocupación con la mayor ocurrencia es: student


11. ¿Cuál es la edad media de los usuarios?

In [57]:
edades = df['age']
print("La edad media de los usuarios es:",edades.mean())

La edad media de los usuarios es: 34.05196182396607


12. ¿Cuál es la edad con menos ocurrencia?

In [113]:
diccionario2 = {}

edades_diferentes = df['age'].unique()

for edad in edades_diferentes:
    diccionario2[edad]=0

for edad1 in diccionario2.keys():
    for edad2 in df['age']:
        if edad1==edad2:
            diccionario2[edad1]+=1 

dataframe2 = pd.DataFrame({
            "edad":diccionario2.keys(),
            "ocurrencia":diccionario2.values()
})

print(dataframe2)

minima_ocurrencia = dataframe2['ocurrencia'].min()

lista_de_edades=[]

for i in diccionario2.keys():
    if diccionario2[i]==minima_ocurrencia:
        lista_de_edades.append(i)
print("Las edades con la menor ocurrencia son:")
for i in lista_de_edades:
    print(i)

    edad  ocurrencia
0     24          33
1     53          12
2     23          28
3     33          26
4     42          21
..   ...         ...
56    10           1
57    73           1
58    58           3
59    69           2
60    70           3

[61 rows x 2 columns]
Las edades con la menor ocurrencia son:
7
66
11
10
73
